src/pages/Account/userName/UsernameSetting.js

```diff
@@ -17,19 +17,28 @@ const CHANGE_USERNAME_MUTATION = gql`
 `
 
 const validateUsername = username => {
-  if (!username || username.length < 3) {
+  if (username && username.length < 3) {
     return 'Username should be at least 3 characters long'
   }
+  if (username && username[0] === "@") {
+    return "@ is not allowed for the first character"
+  }
 }
 
-const UsernameSetting = ({ dispatchNewUsername, username, changeUsername }) => {
+const UsernameSetting = ({ dispatchNewUsername, username, name, changeUsername }) => {
+  let _username = username;
+  if (!username && name) {
+    _username = name.toLowerCase().replace(/ /g,"_");
+  }
+
```

Comment:
```js
const normalizedUsername = username || name.toLowerCase().replace(/ /g,"_");
```

Reply:
There is another case: `!username && !name` so we can't use `||` here.

Reply:
`... = username || (name && name.toLower... )`